presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJdbcQueryBuilderJoinPushdownExtended.java

```diff
@@ -0,0 +1,547 @@
+
+        // Make Additional Predicate. This will have the Join Conditions.
+        // Join condition : "table_alias_1"."integer_col_1" = "table_alias_2"."integer_col_2" AND "table_alias_2."integer_col_2" = "table_alias_3."integer_col_3"
+        StringBuilder predicateAsString = new StringBuilder();
+        predicateAsString.append(quote("table_alias_1")).append(".").append(quote("integer_col_1"));
+        predicateAsString.append(" = ");
+        predicateAsString.append(quote("table_alias_2")).append(".").append(quote("integer_col_2"));
+        predicateAsString.append(" AND ");
+        predicateAsString.append(quote("table_alias_2")).append(".").append(quote("integer_col_2"));
+        predicateAsString.append(" = ");
+        predicateAsString.append(quote("table_alias_3")).append(".").append(quote("integer_col_3"));
+        JdbcExpression predicateAsExpression = new JdbcExpression(predicateAsString.toString());
+        Optional<JdbcExpression> additionalPredicate = Optional.of(predicateAsExpression);
+
+        try (PreparedStatement preparedStatement = new QueryBuilder("\"").buildSql(jdbcClient, session, connection, "test_catalog", "test_schema", "test_table_1", joinTablesList, selectColumns, tupleDomain, additionalPredicate);
+                ResultSet resultSet = preparedStatement.executeQuery()) {
+            ImmutableSet.Builder<Integer> integerCol1Values = ImmutableSet.builder();
+            ImmutableSet.Builder<Integer> integerCol2Values = ImmutableSet.builder();
+            ImmutableSet.Builder<Integer> integerCol3Values = ImmutableSet.builder();
+
+            while (resultSet.next()) {
+                integerCol1Values.add(resultSet.getInt("integer_col_1"));
+                integerCol2Values.add(resultSet.getInt("integer_col_2"));
+                integerCol3Values.add(resultSet.getInt("integer_col_3"));
+            }
+
+            // Expected values based on data setup
+            ImmutableSet<Integer> expectedIntegerCol1Values = ImmutableSet.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+            ImmutableSet<Integer> expectedIntegerCol2Values = ImmutableSet.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+            ImmutableSet<Integer> expectedIntegerCol3Values = ImmutableSet.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+
+            // We are Asserting the preparedStatement and also the resultSet.
+            String expectedPreparedStatement = "SELECT \"table_alias_1\".\"integer_col_1\", \"table_alias_2\".\"integer_col_2\", \"table_alias_3\".\"integer_col_3\" " +
+                    "FROM \"test_schema\".\"test_table_3\" \"table_alias_3\", \"test_schema\".\"test_table_2\" \"table_alias_2\", \"test_schema\".\"test_table_1\" \"table_alias_1\" " +
+                    "WHERE \"table_alias_1\".\"integer_col_1\" = \"table_alias_2\".\"integer_col_2\" AND \"table_alias_2\".\"integer_col_2\" = \"table_alias_3\".\"integer_col_3\"";
+
+            assertTrue(preparedStatement.toString().contains(expectedPreparedStatement),
```

Comment:
While I appreciate the thoroughness of this and other tests in this class, the core of what we want to test is how QueryBuilder is now able to -
1. Use aliases while building the SELECT list
2. Can build a FROM clause that represents a CSV of join tables

These tests can be added as additional tests in `TestJdbcQueryBuilder`.
The rest of what you're testing here - that a query with aliases, FROM clause and join related WHERE clauses in the body is only testing if H2 can support this syntax. This is something we already know it does, so we don't need to _acutally execute the query_ on H2 to prove this